lib/gui/app.js

```diff
@@ -298,11 +298,27 @@ app.controller('HeaderController', function(SelectionStateModel, OSOpenExternalS
 
 });
 
-app.controller('StateController', function($state) {
+app.controller('StateController', function($rootScope) {
+
+  $rootScope.$on('$stateChangeSuccess', (event, toState, toParams, fromState) => {
```

Comment:
This is probably pedantic, but a standard practice is to register for the `$destroy` event of the controller and un-register any registered event.